Goodbye, TEXTSTRING*

rr- opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

Removes the `TEXTSTRING*` stateful API and its leftover usages, replacing it with a lightweight `ui/text.c` module. The new API contains essentially two stateless (but still cached) functions: `UI_Text_Measure` and `UI_Text_Draw` (not counting a small helper for word wrap).

#### Testing

- Various text scaling and screen sizes
- Word wrap behavior for long console output and examine text descriptions
- Text positioning, especially with regard to other elements such as frames or healthbars

rr- committed:
text: remove unused functions

rr- committed:
tr1/savegame: fix compile warning

rr- committed:
text: remove unused properties

rr- committed:
text: tidy constants

rr- committed:
text: move to ui/; simplify

rr- committed:
console: fix valgrind warnings

There were a few bytes leaking when shutting the game down.

aredfan commented:
I did a quick bit of testing with the examine test level. I've not tested anything else yet.

- [x] It looks like the box doesn't have a maximum width for `Key 3`.
<details><summary>Click here to see the screenshot</summary>

![20250523_073446_Caves](https://github.com/user-attachments/assets/93095feb-18a3-4e39-b2bd-046bec35eaaa)

</details>

- [x] `Pickup 1` and `Pickup 2` are missing their respective texts.

<details><summary>Click here to see the screenshot</summary>

![20250523_073533_Caves](https://github.com/user-attachments/assets/f6b1f40b-5c7c-4783-af7e-eb0a6e0a18db)

</details>

rr- committed:
option/examine: fix overlapping with the UI

rr- commented:
> I did a quick bit of testing with the examine test level. I've not tested anything else yet.
> 
> * [ ]  It looks like the box doesn't have a maximum width for `Key 3`.
> 
> Click here to see the screenshot
> * [ ]  `Pickup 1` and `Pickup 2` are missing their respective texts.
> 
> Click here to see the screenshot

I fixed and upgraded it so it doesn't just guess where to break a line based on character count anymore; now it actually checks the available space down to the pixel. Also, I modified the examine item dialog to cover 2/3 of the screen, and made sure it doesn't overlap other UI elements when the text size is large.

![20250523_140951_Caves](https://github.com/user-attachments/assets/ef7cbb4d-255f-4ef5-a080-bed202dedba4)
![20250523_140947_Caves](https://github.com/user-attachments/assets/02593494-964c-4e89-803f-3a9f50d933f0)

![20250523_140626_Laras_Home](https://github.com/user-attachments/assets/fb782b38-0092-4322-b11c-465eab2d4385)

rr- commented:
Hmm… I think the pagination is not working as it should.

rr- commented:
OK – forced page breaks should be fixed as well.

aredfan commented:
> I fixed and upgraded it so it doesn't just guess where to break a line based on character count anymore; now it actually checks the available space down to the pixel. Also, I modified the examine item dialog to cover 2/3 of the screen, and made sure it doesn't overlap other UI elements when the text size is large.

That's very cool, it's great to hear about improvements to the code. 👍 

I can confirm the previous issues are fixed.

_________________

New issue:

- [ ] ~~I noticed the empty lines are no longer appearing for `Key 2` in the examine test level.~~

<details><summary>Click here to see the screenshot</summary>

![20250523_132554_Caves](https://github.com/user-attachments/assets/c0ec8b46-dc96-4d8a-a6ad-673287a13903)

</details>

lahm86 commented:
Looks good. The only difference I noticed with develop is that examine text beginning with new lines is now trimmed. Dev below.

![image](https://github.com/user-attachments/assets/bb69ab4c-ac92-4200-b179-ecfa5d9ea2bc)

rr- commented:
> Looks good. The only difference I noticed with develop is that examine text beginning with new lines is now trimmed. Dev below.
> 
> ![image](https://private-user-images.githubusercontent.com/33758420/447018708-bb69ab4c-ac92-4200-b179-ecfa5d9ea2bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMDc3NTgsIm5iZiI6MTc0ODAwNzQ1OCwicGF0aCI6Ii8zMzc1ODQyMC80NDcwMTg3MDgtYmI2OWFiNGMtYWM5Mi00MjAwLWIxNzktZWNmYTVkOWVhMmJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDEzMzczOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhMWFjOGQ3ZWFiMjY0MzZhM2UxYWVjOTdlMDU3MmFmZTM2MTFiYTVmNjM4ZDYyMGE0ODA5YjgzMzkxZmMyZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.y7_pRnbOyms28CUKOcREaAGTAheuGN0eBZ5-E9KvXks)

This is an intentional change - I don't see a use case for this?

lahm86 commented:
That's fair, would it be worth a changelog mention just in case builders have made use of it?

rr- commented:
> That's fair, would it be worth a changelog mention just in case builders have made use of it?

Thanks. Amended the changelog; will wait for the release prior to merging.
